### ISSUE TITLE
Does not compile with Angular 2 beta 16

### ISSUE THREAD

**Issue by RoopeHakulinen** [shlomiassaf/angular2-modal#65]
Compilation produces the following two errors
```
TypeScript error: /home/vagrant/Documents/eura2014/paatososio/paatososio-web-angular/node_modules/angular2-modal/dist/commonjs/presets/base/ModalAwarePreset.d.ts(1,10): Error TS2305: Module '"/home/vagrant/Documents/eura2014/paatososio/paatososio-web-angular/node_modules/angular2/core"' has no exported member 'ResolvedProvider'.
TypeScript error: /home/vagrant/Documents/eura2014/paatososio/paatososio-web-angular/node_modules/angular2-modal/dist/commonjs/providers/Modal.d.ts(1,46): Error TS2305: Module '"/home/vagrant/Documents/eura2014/paatososio/paatososio-web-angular/node_modules/angular2/core"' has no exported member 'ResolvedProvider'.
```

**Comment by RoopeHakulinen** [comment]
I'll take a look and submit a PR soon, if I get it to work.

**Comment by shlomiassaf** [comment]
@RoopeHakulinen 
I have a local branch in the works for that, there is also a pending PR from @jdelgadoalfonso 

beta16 has a lot of breaking changes... although most breaking changes are domain naming and moving functionality around, I want to investigate it to see if I can improve things.

**Comment by RoopeHakulinen** [comment]
Thanks for the info @shlomiassaf . I took a look on it and it was quite trivial for the other parts, but the change in ElementRef, ViewContainerRef and ComponentRefs with  seems to be quite hard to tackle with the current design. 

Hope you are able to fix it soon as this is really neat module and we are using it extensively in our app, and we can't also upgrade to beta 16 before this is fixed.

**Comment by shlomiassaf** [comment]
@RoopeHakulinen I already have a solution in place, the change is not dramatic...
I just want to see if what I did is reasonable.

Living on beta :) thats the cost

**Comment by RoopeHakulinen** [comment]
Of course it is. Thank you.

**Comment by shlomiassaf** [comment]
@RoopeHakulinen You can see a preview of the work [here](https://github.com/shlomiassaf/angular2-modal/tree/ng2.Beta16-universal.platform-naming.refactor)

This is a big refactor in terms of file structure and file/class renaming.
I took the opportunity to do it, something I wanted to do for a long while.

This will effect people who extend the core functionality.

I have also start to decouple `bootstrap` from the library.
This will enable platforms, bootstrap is a platform, i'v also added `vex` which is a nice dialog JS lib.
there is a sample demo of vex which is of course not ready but I will add functionality soon.

Developers can now start adding other nice dialog's , if they want.
This is still WIP...

**Comment by shlomiassaf** [comment]
Fixed in PR#11
